### PR TITLE
Add diff and check modes for route_maps and prefix_lists modules

### PR DESCRIPTION
**Pull request by mingjunzhang2019**
##### SUMMARY
This is to add diff and check modes support for route_maps and prefix_lists modules.

##### GitHub Issues
N/A

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
sonic_route_maps, sonic_prefix_lists

##### OUTPUT
- route_maps regression result: 
[route_maps_regression-2024-01-26-10-33-55.html.pdf](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/14090282/route_maps_regression-2024-01-26-10-33-55.html.pdf)
- route_maps regression log: 
[route_maps_regression.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/14090289/route_maps_regression.log)
- route_maps unit test log: 
[route_maps_ut.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/14090292/route_maps_ut.log)
- route_maps unit test script: 
[route_maps_ut.txt](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/14090294/route_maps_ut.txt)

- prefix_lists regression result: 
[prefix_lists_regression-2024-01-29-13-47-15.html.pdf](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/14090296/prefix_lists_regression-2024-01-29-13-47-15.html.pdf)
- prefix_lists regression log: 
[prefix_lists_regression.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/14090299/prefix_lists_regression.log)
- prefix_lists unit test log: 
[prefix_lists_ut.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/14090301/prefix_lists_ut.log)
- prefix_lists unit test script: 
[prefix_lists_ut.txt](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/14090303/prefix_lists_ut.txt)

##### Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility or have provided any relevant "breaking_changes" descriptions in a "fragment" file in the "changelogs/fragments" directory of this repository.
- [x] I have provided a summary for this PR in valid "fragment" file format in the "changelogs/fragments" directory of this repository branch. Reference : [Ansible Change Log Document](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)

##### How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

did regression testing and lots of unit testing.

### CODE REVIEW

File: plugins/modules/sonic_prefix_lists.py [hunk 1]
```diff
@@ -492,6 +494,34 @@
...
+  type: list
+  sample: >
+    The configuration returned will always be in the same format
+     of the parameters above.
+commands:
+  description: The set of commands pushed to the remote device.
+  returned: always
+  type: list
+  sample: ['command 1', 'command 2', 'command 3']
+"""
```
Comment (kerry-meyer) [inline plugins/modules/sonic_prefix_lists.py:524]: It looks like we should probably also have some description for the "diff" output. The same issue applies to route_maps as well. (This was probably missed for other check/diff PRs. If necessary, we can have a separate PR to add it to those resource modules.)

**Review by kerry-meyer — CHANGES_REQUESTED** [review]
The changes and test results look good.

I am posting a couple of minor comments to resolve before we merge this PR.

File: plugins/module_utils/network/sonic/config/route_maps/route_maps.py [hunk 2]
```diff
@@ -2352,3 +2376,19 @@ def validate_and_normalize_config(self, input_config_list):
                 route_map['match']['peer']['interface'] = updated_intf_name
 
         return updated_config_list
+
+    def sort_lists_in_config(self, config):
+        if config:
+            config.sort(key=lambda x: (x.get('map_name', None) is not None,
```
[the remark this answers is not in the document]
Comment (mingjunzhang2019) [reply plugins/module_utils/network/sonic/config/route_maps/route_maps.py]: Changed. Thanks!

**Comment by mingjunzhang2019** [comment]
Merged with main branch!

File: plugins/modules/sonic_prefix_lists.py (hunk 1 above)
Comment (mingjunzhang2019) [reply plugins/modules/sonic_prefix_lists.py:531]: Added sample texts.

**Review by kerry-meyer — APPROVED** [review]
The updated changes and posted test results all look good.

Thank you for providing this enhancement to our diff/check functionality.

Approved.

**Pull request merged**